models/migrations/v214.go

```diff
@@ -0,0 +1,248 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"code.gitea.io/gitea/models/auth"
+	"code.gitea.io/gitea/models/perm"
+	"code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/models/user"
+	api "code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"xorm.io/xorm"
+)
+
+func convertFromNullToDefault(x *xorm.Engine) error {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	type Label struct {
+		ID              int64 `xorm:"pk autoincr"`
+		RepoID          int64 `xorm:"INDEX"`
+		OrgID           int64 `xorm:"INDEX"`
+		Name            string
+		Description     string
+		Color           string             `xorm:"VARCHAR(7)"`
+		NumIssues       int                `xorm:"NOT NULL DEFAULT 0"`
+		NumClosedIssues int                `xorm:"NOT NULL DEFAULT 0"`
+		CreatedUnix     timeutil.TimeStamp `xorm:"INDEX created"`
+		UpdatedUnix     timeutil.TimeStamp `xorm:"INDEX updated"`
+	}
+	recreateTable(sess, &Label{})
```

Comment:
do we have to recreate the table?
for sqlite sure but for other DBMS we dont

Reply:
See discussion on top.